pex/cli/commands/lock.py

```diff
@@ -1002,23 +1000,42 @@ def _create_lock_update_request(
         )
 
         with TRACER.timed("Selecting locks to update"):
-            subset_result = try_(
-                subset(
-                    targets=targets,
-                    lock=lock_file,
-                    network_configuration=network_configuration,
-                    build_configuration=lock_file.build_configuration(),
-                    transitive=lock_file.transitive,
+            locked_resolve_count = len(lock_file.locked_resolves)
+            if lock_file.style is LockStyle.UNIVERSAL and locked_resolve_count != 1:
+                return Error(
+                    "The lock at {path} contains {count} locked resolves; so it "
+                    "cannot be updated as a universal lock which requires exactly one locked "
+                    "resolve.".format(path=lock_file_path, count=locked_resolve_count)
                 )
-            )
-
-        update_requests = [
-            ResolveUpdateRequest(
-                target=resolved_subset.target, locked_resolve=resolved_subset.resolved.source
-            )
-            for resolved_subset in subset_result.subsets
-        ]
-        if self.options.strict and lock_file.style is not LockStyle.UNIVERSAL:
+            if locked_resolve_count == 1:
+                locked_resolve = lock_file.locked_resolves[0]
+                update_targets = (
+                    [LocalInterpreter.create(targets.interpreter)]
+                    if lock_file.style is LockStyle.UNIVERSAL
+                    else targets.unique_targets()
+                )
+                update_requests = [
+                    ResolveUpdateRequest(target=target, locked_resolve=locked_resolve)
+                    for target in update_targets
+                ]
+            else:
+                subset_result = try_(
```

Comment:
N.B.: This is the old logic and it still applies for lock files with more than one locked resolve within; so these locks will still be subject to the more strict locked resolve selection rules and potentially fail to sync. I'll add a comment here pointing to an issue that tracks possibly handling those style locks better with a less strict, but still correct, scheme for selecting locked resolves to sync.

Reply:
Comment added and filed https://github.com/pex-tool/pex/issues/2386